src/views/ProposalReviewUserDetail.vue

```diff
@@ -0,0 +1,163 @@
+          <basic-custom-field
+            v-model="form.comments"
+            :errors="apiValidationErrors.comments"
+            placeholder="Enter your comments here..."
+            label="Comments"
+            field="comments"
+            field-type="textarea"
+          ></basic-custom-field>
+
+          <basic-custom-field
+            v-model="form.grade"
+            :errors="apiValidationErrors.grade"
+            label="Grade"
+            field="grade"
+            type="number"
+            min="0"
+            max="10"
+            step="0.01"
+            description="Between 0.00 and 10.00, where higher is better. E.g. 9.25"
+            placeholder="Enter grade"
+          ></basic-custom-field>
+
+          <basic-custom-field
+            v-model="form.finished"
+            field-type="checkbox"
+            field="finished"
+            description="Check when you have finished making changes."
+            :errors="apiValidationErrors.finished"
+            checkbox-option-label="Finished"
+            value=true
+          ></basic-custom-field>
+
+          <b-button type="submit" variant="primary" :disabled="submitDisabled">Save</b-button>
+        </b-form>
+        </b-card-text>
+      </b-card>
+    </b-container>
+  </ocs-data-loader>
+</template>
+<script>
+import $ from 'jquery';
+
+import { OCSMixin } from 'ocs-component-lib';
+
+import NotFound from '@/components/NotFound.vue';
+import BasicCustomField from '@/components/util/BasicCustomField.vue';
+
+export default {
+  name: 'ProposalReviewUserDetail',
+  components: {
+    NotFound,
+    BasicCustomField
+  },
+  mixins: [OCSMixin.getDataMixin],
+  props: {
+    id: {
+      type: String,
+      required: true
+    }
+  },
+  data: function() {
+    return {
+      form: {
+        comments: "",
+        grade: null,
+        finished: false
+      },
+      submitDisabled: false,
+      apiValidationErrors: {}
+    };
+  },
+  watch: {
+    data: function(data) {
+      this.form.comments = data.my_review?.comments;
+      this.form.grade = data.my_review?.grade;
+      this.form.finished = data.my_review?.finished;
+    }
+  },
+  methods: {
+    initializeDataEndpoint: function() {
+      return this.$store.state.urls.observationPortalApi + '/api/scienceapplication-reviews/' + this.id;
+    },
+    onSubmit(event) {
+      event.preventDefault();
+      this.submitDisabled = true
+
+      const that = this;
+
+      $.ajax({
+        method: this.data.my_review === null ? "POST" : "PUT",
+        url: this.initializeDataEndpoint() + "/my-review",
+        data: JSON.stringify(this.form),
+        dataType: "json",
+        processData: "false",
+        contentType: "application/json",
+        cache: false
+      }).done(function(response) {
+        that.apiValidationErrors = {};
+
+        that.data.my_review = response;
+        that.$store.commit("addMessage", { text: "Saved", variant: "success" });
+
+        that.dataNotFound = false;
```

Comment:
This is up for debate but I kind of think after saving it should route you back to the proposal reviews table page... At least when you save after selecting finished maybe?

Reply:
Going to keep it simple for now. If people find it cumbersome to navigate back and forth, we can add it then.